src/braket/aws/aws_session.py

```diff
@@ -645,6 +647,8 @@ def search_devices(
                     continue
                 if statuses and result["deviceStatus"] not in statuses:
                     continue
+                if statuses is None and result["deviceStatus"] == "RETIRED":
+                    continue
```

Comment:
Am I missing something, or does this not actually impact the `search_devices` call? Does this fix our time issue? It looks like it still gets every device but then filters after; is this a limitation of the search devices api?

Reply:
It does fix our time issue. Even though the API returns all values, we will filter our the retired ones. This means that when we actually do get_device for all returned devices, we will be doing many fewer calls.

Reply:
Ah that's the piece I was missing, that makes sense if `get_device` is what's taking a long time